shared/core/types/param.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+import (
+	"fmt"
+	"github.com/pokt-network/pocket/shared/crypto"
+)
+
+func (p Param) Hash() []byte {
```

Comment:
In `codec.go`, the marshal function looks like so (note the `Deterministic: true`):

```go
func (p *ProtoCodec) Marshal(message proto.Message) ([]byte, error) {
	bz, err := proto.MarshalOptions{Deterministic: true}.Marshal(message)
	if err != nil {
		return nil, err
	}
	return bz, nil
}
```

Rather than "rolling our own" serialization for the key, what if the hash is just the sha of the serialized proto?

```go
paramBz, err := codec.GetCodec().Marshal(param)
paramKey := crypto.SHA3Hash(paramBz)
```

This should still be deterministic and result in less proprietary business logic.

Reply:
@Olshansk 🤦🏼‍♂️ This is much cleaner ahah